ArduCopter/mode_autorotate.cpp

```diff
@@ -85,24 +101,61 @@ void ModeAutorotate::run()
     // Current time
     uint32_t now = millis(); //milliseconds
 
+    // set dt in library
+    float const last_loop_time_s = AP::scheduler().get_last_loop_time_s();
+    g2.arot.set_dt(last_loop_time_s);
+
+    float alt = g2.arot.get_ground_distance();
+    // have autorotation library update estimated radar altitude
+    g2.arot.update_est_radar_alt();
+    if (alt < copter.rangefinder.max_distance_cm_orient(ROTATION_PITCH_270)) {
+        g2.arot._using_rfnd = true;
+    } else {
+        g2.arot._using_rfnd = false;
+    }
+
     // Initialise internal variables
     float curr_vel_z = inertial_nav.get_velocity_z_up_cms();   // Current vertical descent
+    //calculate time to impact
+    if (phase_switch != Autorotation_Phase::TOUCH_DOWN) {
+        g2.arot.time_to_ground();
+        time_to_impact = g2.arot.get_time_to_ground();
+        last_tti=time_to_impact;
+    } else {
+        time_to_impact = last_tti;
```

Comment:
I think `time_to_impact` could be a float local to this function rather than needing to be global to the class. I don't think its used anywhere else?

Reply:
done